test/vanilla/ignite-generate.test.ts

```diff
@@ -0,0 +1,164 @@
+        "   
+           
+           Generated new files:
+           /user/home/ignite/app/models/PizzaStore.test.ts
+           /user/home/ignite/app/models/PizzaStore.ts
+        "
+      `)
+      expect(read(`${TEMP_DIR}/app/models/PizzaStore.ts`)).toMatchInlineSnapshot(`
+        "import { Instance, SnapshotIn, SnapshotOut, types } from \\"mobx-state-tree\\"
+
+        /**
+         * Model description here for TypeScript hints.
+         */
+        export const PizzaStoreModel = types
+          .model(\\"PizzaStore\\")
+          .props({})
+          .views((self) => ({})) // eslint-disable-line @typescript-eslint/no-unused-vars
+          .actions((self) => ({})) // eslint-disable-line @typescript-eslint/no-unused-vars
+
+        export interface PizzaStore extends Instance<typeof PizzaStoreModel> {}
+        export interface PizzaStoreSnapshotOut extends SnapshotOut<typeof PizzaStoreModel> {}
+        export interface PizzaStoreSnapshotIn extends SnapshotIn<typeof PizzaStoreModel> {}
+        export const createPizzaStoreDefaultModel = () => types.optional(PizzaStoreModel, {})
+        "
+      `)
+      expect(read(`${TEMP_DIR}/app/models/PizzaStore.test.ts`)).toMatchInlineSnapshot(`
+        "import { PizzaStoreModel } from \\"./PizzaStore\\"
+
+        test(\\"can be created\\", () => {
+          const instance = PizzaStoreModel.create({})
+
+          expect(instance).toBeTruthy()
+        })
+        "
+      `)
+      expect(read(`${TEMP_DIR}/app/models/index.ts`)).toMatchInlineSnapshot(`
+        "export * from \\"./RootStore\\"
+        export * from \\"./helpers/get-root-store\\"
+        export * from \\"./helpers/use-stores\\"
+        export * from \\"./helpers/setup-root-store\\"
+        export * from \\"./PizzaStore\\"
+        "
+      `)
+      expect(read(`${TEMP_DIR}/app/models/RootStore.ts`)).toMatchInlineSnapshot(`
+        "import { Instance, SnapshotOut, types } from \\"mobx-state-tree\\"
+        import { PizzaStoreModel } from \\"./PizzaStore\\"
+        import { AuthenticationStoreModel } from \\"./AuthenticationStore\\" // @demo remove-current-line
+        import { EpisodeStoreModel } from \\"./EpisodeStore\\" // @demo remove-current-line
+
+        /**
+         * A RootStore model.
+         */
+        export const RootStoreModel = types.model(\\"RootStore\\").props({
+          pizzaStore: types.optional(PizzaStoreModel, {} as any),
+          authenticationStore: types.optional(AuthenticationStoreModel, {}), // @demo remove-current-line
+          episodeStore: types.optional(EpisodeStoreModel, {}), // @demo remove-current-line
+        })
+
+        /**
+         * The RootStore instance.
+         */
+        export interface RootStore extends Instance<typeof RootStoreModel> {}
+        /**
+         * The data of a RootStore.
+         */
+        export interface RootStoreSnapshot extends SnapshotOut<typeof RootStoreModel> {}
+        "
+      `)
```

Comment:
I think for this if we are going to get so specific we should just do a regular snapshot instead.

I'd probably just check for `export const RootStoreModel` and call it good, unless there's a specific reason we need to test all the other details.

Reply:
So the nice thing about this, is that Jest automatically creates this snapshot: https://jestjs.io/docs/snapshot-testing#inline-snapshots

It's the same as `toMatchSnapshot`, but the snapshot is colocated

Reply:
And the snapshot can be updated by pressing `u` in the test runner

Reply:
Main thing I like about snapshots or things like this is that we get more detailed error messages when a test throws

Reply:
Here is a video of what I'm talking about

https://user-images.githubusercontent.com/37849890/190252650-8603ffda-e0ef-4d8c-96aa-0fca7d599b27.mov